tests/models/test_dag.py

```diff
@@ -387,243 +388,93 @@ def test_get_num_task_instances(self):
         self.assertEqual(
-            expected.var1,
-            actual.var1,
-            msg="Named tuples may not have been preserved in rendering"
-        )
-        self.assertEqual(
-            expected.var2,
-            actual.var2,
-            msg="Named tuples may not have been preserved in rendering"
-        )
-
-    def test_render_template_dict_field(self):
-        """Tests if render_template from a dict field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
-        self.assertDictEqual(
-            task.render_template('', {'key1': '{{ foo }}_1', 'key2': '{{ foo }}_2'}, {'foo': 'bar'}),
-            {'key1': 'bar_1', 'key2': 'bar_2'}
-        )
-
-    def test_render_template_dict_field_with_templated_keys(self):
-        """Tests if render_template from a dict field works as expected:
-        dictionary keys are not templated"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
-        self.assertDictEqual(
-            task.render_template('', {'key_{{ foo }}_1': 1, 'key_2': '{{ foo }}_2'}, {'foo': 'bar'}),
-            {'key_{{ foo }}_1': 1, 'key_2': 'bar_2'}
-        )
-
-    def test_render_template_date_field(self):
-        """Tests if render_template from a date field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
-        self.assertEqual(
-            task.render_template('', datetime.date(2018, 12, 6), {'foo': 'bar'}),
-            datetime.date(2018, 12, 6)
+            expected.var1, actual.var1, msg="Named tuples may not have been preserved in rendering"
         )
-
-    def test_render_template_datetime_field(self):
-        """Tests if render_template from a datetime field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
-            task = DummyOperator(task_id='op1')
-
         self.assertEqual(
-            task.render_template('', datetime.datetime(2018, 12, 6, 10, 55), {'foo': 'bar'}),
-            datetime.datetime(2018, 12, 6, 10, 55)
+            expected.var2, actual.var2, msg="Named tuples may not have been preserved in rendering"
         )
 
     def test_render_template_UUID_field(self):
         """Tests if render_template from a UUID field works"""
-
-        dag = DAG('test-dag',
-                  start_date=DEFAULT_DATE)
-
-        with dag:
+        with DAG('test-dag', start_date=DEFAULT_DATE):
             task = DummyOperator(task_id='op1')
 
         random_uuid = uuid.uuid4()
-        self.assertIs(
-            task.render_template('', random_uuid, {'foo': 'bar'}),
-            random_uuid
-        )
+        self.assertIs(task.render_template(random_uuid, {'foo': 'bar'}), random_uuid)
```

Comment:
Same here (add in to the parameterized list? Harder since this is random?)

Reply:
Same here, we must call `random_uuid = uuid.uuid4()` in the test, and thus it doesn't "fit" the general parameterized test.